### ISSUE TITLE
Unable to determine granted permission for assembly.

### ISSUE THREAD

**Issue by sibartlett** [andrewdavey/cassette#421]
Installed VS 2013 and .NET 4.5.1, and now get following exception when using the latest NuGet package:

[IsolatedStorageException: Unable to determine granted permission for assembly.]
   System.IO.IsolatedStorage.IsolatedStorage.InitStore(IsolatedStorageScope scope, Type domainEvidenceType, Type assemblyEvidenceType) +622
   System.IO.IsolatedStorage.IsolatedStorageFile.GetStore(IsolatedStorageScope scope, Type domainEvidenceType, Type assemblyEvidenceType) +77
   System.Lazy`1.CreateValue() +812
   System.Lazy`1.LazyInitValue() +476
   Cassette.Aspnet.IsolatedStorageContainer.get_IsolatedStorageFile() +94
   Cassette.IO.IsolatedStorageFile.get_Exists() +19
   Cassette.Aspnet.WebHostSettingsConfiguration.IsStaticCacheManifest(CassetteSettings settings) +86
   Cassette.Aspnet.WebHostSettingsConfiguration.Configure(CassetteSettings settings) +247
   Cassette.ConfigurationEnumerableExtensions.Configure(IEnumerable`1 configurations, T configurable) +176

[TargetInvocationException: Exception has been thrown by the target of an invocation.]
   System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor) +0
   System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture) +748
   Cassette.TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) +873

[TinyIoCResolutionException: Unable to resolve type: Cassette.CassetteSettings]
   Cassette.TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) +1241
   Cassette.TinyIoC.SingletonFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) +165
   Cassette.TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) +1376
   Cassette.TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) +730

[TinyIoCResolutionException: Unable to resolve type: Cassette.FileSystemWatchingBundleRebuilder]
   Cassette.TinyIoC.TinyIoCContainer.ConstructType(Type requestedType, Type implementationType, ConstructorInfo constructor, NamedParameterOverloads parameters, ResolveOptions options) +1049
   Cassette.TinyIoC.SingletonFactory.GetObject(Type requestedType, TinyIoCContainer container, NamedParameterOverloads parameters, ResolveOptions options) +165
   Cassette.TinyIoC.TinyIoCContainer.ResolveInternal(TypeRegistration registration, NamedParameterOverloads parameters, ResolveOptions options) +1376
   System.Linq.WhereSelectEnumerableIterator`2.MoveNext() +265
   System.Linq.<CastIterator>d__b1`1.MoveNext() +296
   Cassette.HostBase.RunStartUpTasks() +248
   Cassette.HostBase.Initialize() +320
   Cassette.Aspnet.CassetteHttpModule.InitWithTraceLogging() +178
   Cassette.Aspnet.CassetteHttpModule.Init(HttpApplication httpApplication) +149
   System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers) +1777
   System.Web.HttpApplication.RegisterEventSubscriptionsWithIIS(IntPtr appContext, HttpContext context, MethodInfo[] handlers) +2504
   System.Web.HttpApplication.InitSpecial(HttpApplicationState state, MethodInfo[] handlers, IntPtr appContext, HttpContext context) +203
   System.Web.HttpApplicationFactory.GetSpecialApplicationInstance(IntPtr appContext, HttpContext context) +389
   System.Web.Hosting.PipelineRuntime.InitializeApplication(IntPtr appContext) +409

[HttpException (0x80004005): Unable to resolve type: Cassette.FileSystemWatchingBundleRebuilder]
   System.Web.HttpRuntime.FirstRequestInit(HttpContext context) +646
   System.Web.HttpRuntime.EnsureFirstRequestInit(HttpContext context) +99
   System.Web.HttpRuntime.ProcessRequestNotificationPrivate(IIS7WorkerRequest wr, HttpContext context) +669

**Comment by sibartlett** [comment]
Just found this Google Group thread: https://groups.google.com/forum/#!topic/cassette/VHzXyxTO2sE

**Comment by SeriousM** [comment]
I got this now as well. very unpleasant...
@andrewdavey: is there an ETA on this problem?
Thanks

**Comment by robpvn** [comment]
This is a regression of issue #154, there's more info there.

**Comment by robpvn** [comment]
It looks like the culprit commit is commit 7fd97a86723a8269c8d403e16acdbf3aae1e0b6a, where the refactoring subtly turns the lambda that executes the function into a lambda that returns a methodgroup for execution. Reverting that single commit should be enough to fix this.

**Comment by robpvn** [comment]
Actually, I may have read that wrong. It seems on closer inspection that they should be functionally equal. Then it could also be 78487696b9baa163e6a110f86b6f3c874e1e5866 where the static attribute is removed, that might be complicating things for the compiler even though I'm not sure about this. Problem is that it can't simply be reverted because the changes appear to have been made to fix another issue.

**Comment by andrewdavey** [comment]
So far I've been unable to reproduce this exception. I created a new web project in VS 2013, targeting .NET 4.5.1. I ran it using IIS Express and it seems work just fine.

Can someone let me know what configuration/server/etc they are running on please?

**Comment by robpvn** [comment]
In our case it works fine when locally on IIS Express and .Net 4.5, but when deployed to a server with IIS 8 it malfunctions.

**Comment by vondruska** [comment]
Running into this exception using IIS 7.5 with .NET 4.5. I'm with @robpvn, everything runs great on IIS Express.

**Comment by andrewdavey** [comment]
Thanks for the configuration info guys. Can you give above pull request a test please? (I'm still unable to reproduce the bug on my local Windows 7 IIS, so can't confirm.)

**Comment by jamescrowley** [comment]
@andrewdavey we've just had all our prod servers go down with this error running Cassette 2.2.1. It _appears_ at this time to be related to security update KB2898865 which was applied last night. Not sure if related to previous occurrences of this error or not.

**Comment by jamescrowley** [comment]
@andrewdavey Uninstalling the security update has fixed for now, but obviously not a long term solution. BTW Our servers are Win 2012, .NET 4.5 (_not_ 4.5.1)

**Comment by jamescrowley** [comment]
I don't suppose anyone else affected by this issue is using NewRelic on their servers?

**Comment by SeriousM** [comment]
Even I'm using NR on my prod server, I got the error on my local machine. I
don't think NR is a part of this issue.

**Comment by nazjunaid** [comment]
@SeriousM  @jamescrowley Have you tried #441 it has fixed the issue for us when upgrading to 2.4.0

**Comment by SeriousM** [comment]
> @SeriousM @jamescrowley Have you tried #441 it has fixed the issue for us when upgrading to 2.4.0

nope, I left the project so I can't test it anymore.

**Comment by robinminto** [comment]
@jamescrowley I encountered this issue when I installed New Relic. Disabling the app pool instrumentation removed the issue. Non-Cassette apps are instrumented by New Relic without problems. This is using Cassette 2.0.0 on Windows 2008 R2. I've applied the patch in KB2898869 (which I think is the Win2008 equivalent of KB2898865 for Win2012) without problems.

**Comment by vondruska** [comment]
@andrewdavey looks like #441 fixes the issue for me

**Comment by andrewdavey** [comment]
v2.4.1 has a fix for this

**Comment by romanko89** [comment]
@robinminto I had the same problem, despite I am using **v 2.4.2.** Uninstalling of NewRelic client fixed issue. Then later I saw your advice to change app pool settings - haven't tried it.

**Comment by japf** [comment]
@romanko89 I'm my case adding _cacheDirectory="App_Data\Cassette"_ to the <cassette> node in the Web.config as a workaround solved the issue.

**Comment by romanko89** [comment]
hi, @japf thanks, I'll try your solution, If I need using NewRelic again.

**Comment by jaymcguinness** [comment]
Sorry to comment on an old issue - but in case it helps anyone else:

We support a site built a number of years ago, running Cassette 2.0.0.  We installed the NewRelic .Net Agent yesterday - and this morning, we got this error.  So it does appear to be related to a combination NewRelic and older versions of Cassette.

Switching from 64bit to 32bit app pool got us back up and running in the short term.
